ateam_kenobi/src/in_play_eval.hpp

```diff
@@ -0,0 +1,131 @@
+  // optional in case ball cant be seen
+  std::optional<ateam_geometry::Point> maybe_kickoff_position {};
+
+  static constexpr double DIST_THRESHOLD {0.05};
+
+  // only 3 things enter us into play according to the appendix B state machine
+  void update(ateam_kenobi::World & world)
+  {
+    // edge triggered
+    // we just need an edge on commands to know not to do in_play = false in the same normal_play
+    ateam_common::GameCommand next_command = world.referee_info.running_command;
+    if (cur_state != next_command) {
+      switch (next_command) {
+        // all of these are not in play technically but not for this purpose
+        case ateam_common::GameCommand::Halt:
+        case ateam_common::GameCommand::Stop:
+        case ateam_common::GameCommand::TimeoutTheirs:
+
+        case ateam_common::GameCommand::PrepareKickoffOurs:
+        case ateam_common::GameCommand::PrepareKickoffTheirs:
+        case ateam_common::GameCommand::DirectFreeOurs:
+        case ateam_common::GameCommand::DirectFreeTheirs:
+        case ateam_common::GameCommand::BallPlacementOurs:
+        case ateam_common::GameCommand::BallPlacementTheirs:
+        case ateam_common::GameCommand::NormalStart:
+          // reset in_play in any of these state
+          world.in_play = false;
+          break;
+
+        case ateam_common::GameCommand::PreparePenaltyOurs:
+          world.in_play = false;
+          world.our_penalty = true;
+          world.their_penalty = false;
+          break;
+        case ateam_common::GameCommand::PreparePenaltyTheirs:
+          world.in_play = false;
+          world.their_penalty = true;
+          world.our_penalty = false;
+          break;
+
+        // force play really play acts the same as play
+        case ateam_common::GameCommand::ForceStart:
+          world.in_play = true;
+          world.their_penalty = false;
+          world.our_penalty = false;
+          break;
+        default:
+          world.in_play = true;
+          world.their_penalty = false;
+          world.our_penalty = false;
+      }
+    }
+
+    cur_state = next_command;
+
+    // level
+    if (in_play == false) {
+      auto ball_current = world.ball;
+      if (maybe_kickoff_position.has_value()) {
+        // if we are tracking a start position and a current ball and its further away than
+        // dist we are in play
+        if (sqrt(CGAL::squared_distance(ball_current.pos, maybe_kickoff_position.value())) >
+          DIST_THRESHOLD)
+        {
+          world.in_play = true;
+          // so if you read this and its any of the states in the above you are basically
+          // playing under restrictions
+        }
+      } else {
+        // set the ball position if we dont have one yet during this kickoff
+        maybe_kickoff_position = ball_current.pos;
+      }
+    } else {
+      // note we may not have a view of the ball the frame we enter kickoff thats
```

Comment:
Below resets the kick off position before the next instance where we go out of play? Is that the actual purpose?

Reply:
This seems to reset the tracked kickoff position when the ball is in play so that the next time we start tracking whether the ball's in play we don't accidentally carry over state from this time. Seems reasonable.